### PR TITLE
Remove absolute path from build/main.js

### LINKED ISSUE #116: Absolute paths of build server are baked into javascript files
When generating a prod version the absolute path of the build server is baked into the resulting javascript files, so manifest.json and assets.json could not be loaded

### PR DESCRIPTION
**Pull request by fastner**
Fixes #116. This PR should be tested against already existing projects to see if there is no side effect of this change. The purpose of this PR is to make build folder more or less self contained (public is another folder that seems to be necessary). Also this PR solves the problem that CI server usually have checkout point than the final server so writing absolute paths during build process is not a good idea.

As a side effect all eslint errors of ConfigFactory are fixed or disabled where appropriate.

### CODE REVIEW

File: src/server/config.js [hunk 1]
```diff
@@ -3,26 +3,31 @@
 // respective "process.env.*" variables below with the actual values.
 // These config values then become a part of the server bundle.
 
-import { resolve } from "path"
+import { resolve, dirname } from "path"
+
+export const APP_ROOT = resolve(
+  dirname(process.mainModule.filename),
```
Comment (fastner) [inline src/server/config.js]: process.mainModule is used here instead of require.main as the second one is modified by webpack

**Comment by swernerx** [comment]
I don't like adding a APP_ROOT variable. This should be possible to auto detect either by a custom logic e.g. 

1. CWD == root (which works good enough when executed through `npm`)
2. root == queried by app-root-dir: https://www.npmjs.com/package/app-root-dir

**Comment by swernerx** [comment]
I'd prefer this PR if it does not touch a lot of other totally unrelated lines. Can you cleanup unrelated changes?

**Comment by fastner** [comment]
These changes are due to commit hook preventing commits that don't match eslint rules. So we maybe should first clean up all files. I will rework this PR so it gets a lot more lightweight sticking to the general project structure mentioned in the readme file.

**Pull request closed without merging**